lib/ui/painting/image_descriptor.h

```diff
@@ -0,0 +1,89 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_LIB_UI_PAINTING_IMAGE_DESCRIPTOR_H_
+#define FLUTTER_LIB_UI_PAINTING_IMAGE_DESCRIPTOR_H_
+
+#include <cstdint>
+#include <memory>
+
+#include "flutter/fml/macros.h"
+#include "flutter/lib/ui/dart_wrapper.h"
+#include "flutter/lib/ui/painting/immutable_buffer.h"
+#include "third_party/skia/include/codec/SkCodec.h"
+#include "third_party/skia/include/core/SkImageInfo.h"
+#include "third_party/tonic/dart_library_natives.h"
+
+namespace flutter {
+
+class ImageDescriptor : public RefCountedDartWrappable<ImageDescriptor> {
```

Comment:
I think that's confusing given we already have `Codec` and `SingleFrameCodec`/`MultiFrameCodec`. It's also a bit confusing to have `ImageCodec::instantiateCodec`.

I can add the docs. I can't think of a good way to make this object more light weight than it is, or to avoid retaining the underlying data for it.